src/main/java/org/spongepowered/api/attribute/AttributeHolder.java

```diff
@@ -0,0 +1,106 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.attribute;
+
+import com.google.common.base.Optional;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Represents some kind of object that can have {@link Attribute}s assigned to
+ * it that can be modified by {@link AttributeModifier}s.
+ */
+public interface AttributeHolder {
+
+    /**
+     * Calculates the value of a certain {@link Attribute} on this holder.
+     * 
+     * @param attribute The Attribute to find the value of
+     * @return The value of a certain Attribute on this holder, or
+     *         Optional.absent() if the attribute is not applicable
+     */
+    Optional<Double> getAttributeValue(Attribute attribute);
+
+    /**
+     * Sets the base value of a certain {@link Attribute} on this holder.
+     * 
+     * @param attribute The Attribute to set the base value of
+     * @param base The new base value
+     * @throws IllegalArgumentException If the base value exceeds the maximum or
+     *         is below the minimum of the {@link Attribute}
+     */
+    void setBase(Attribute attribute, double base) throws IllegalArgumentException;
+
+    /**
+     * Gets the base value of a certain {@link Attribute} on this holder.
+     * 
+     * @param attribute The attribute to get the base value of
+     * @return The base value of a certain Attribute on this holder
+     */
+    double getBase(Attribute attribute);
+
+    /**
+     * Gets a collection of all applied modifiers.
+     * 
+     * @return A collection all applied modifiers
+     */
+    Collection<AttributeModifier> getAllModifiers();
+
+    /**
+     * Adds a {@link AttributeModifier} to this holder.
+     * 
+     * @param modifier The AttributeModifier to add
+     * @throws IllegalArgumentException If the AttributeModifier's
+     *         {@link Attribute}'s trget cannot apply to this holder
+     */
+    void addModifier(AttributeModifier modifier) throws IllegalArgumentException;
```

Comment:
typo `trget` -> `target`
Please state the behaviour when adding it multiple times.
Minecraft throws an `IllegalArgumentException("Modifier is already applied on this attribute!")` in that case.

In addition to that it should be be stated in AttributeModifier class's javadocs, that `AttributeModifier`s can be used multiple times, for the same purpose. Otherwise plugin authors will create a new instance each time, they want to use it and will fail to remove it because they cannot remove the old modifier with the new one.